bin/utils/utils.py

```diff
@@ -0,0 +1,350 @@
+        capture_output=True,
+    )
+    output_lines = result.stdout.decode("utf-8").splitlines()
+    image_names = set(
+        [
+            line_parts[1]
+            for line_parts in [line.split() for line in output_lines]
+            if len(line_parts) == 2 and line_parts[0].startswith(PROJECT_NAME_PREFIX)
+        ]
+    )
+
+    if len(image_names) == 0:
+        raise D2DockerError("There are no d2-docker images running")
+    elif len(image_names) == 1:
+        image_name = list(image_names)[0]
+        logger.info("Image is running: {}".format(image_name))
+        return image_name
+    else:
+        names = "\n".join("  " + s for s in image_names)
+        raise D2DockerError("Multiple d2-docker images running, specify one:\n{}".format(names))
+
+
+def get_image_status(image_name, first_port=8080):
+    """
+    If the container for the image is not running, return:
+
+        {
+            "state": "stopped"
+        }
+
+    If it's running, return:
+
+        {
+            "state": "running",
+            "containers": {
+                "core": CONTAINER_CORE,
+                "gateway": CONTAINER_GATEWAY,
+                "db": CONTAINER_DB,
+            },
+            "port": PORT,
+        }
+    Return a Result(status=True, value=value) available port for an image.
+    If it's already running, return Result(status=False, error=error).
+    """
+    final_image_name = image_name or get_running_image_name()
+    project_name = get_project_name(final_image_name)
+    result = run(["docker", "ps", "--format={{.Names}} {{.Ports}}"], capture_output=True)
+    output_lines = result.stdout.decode("utf-8").splitlines()
+
+    containers = {}
+    port = None
+
+    for line in output_lines:
+        parts = line.split(None, 1)
+        container_name, ports = parts
+        if container_name.startswith(project_name + "_"):
+            service = container_name[len(project_name) + 1 :].split("_", 1)[0]
+            containers[service] = container_name
+            if service == "gateway":
+                port = get_port_from_docker_ports(ports)
+
+    if containers and port:
+        return {"state": "running", "containers": containers, "port": port}
+    else:
+        return {"state": "stopped"}
+
+
+def get_port_from_docker_ports(info):
+    port_re = r"0\.0\.0\.0:(\d+)->"
+    match = re.match(port_re, info)
+    port = int(match.group(1)) if match else None
+    return port
+
+
+DHIS2_DATA_IMAGE = "dhis2-data"
+
+
+def get_project_name(image_name):
+    """
+    Return the project name prefix from an image name.
+
+    Example: eyeseetea/dhis2-data:2.30-ento -> eyeseetea-2-30-ento
+
+    The final containers created have names: PROJECT_NAME_{SERVICE}_1.
+    """
+    clean_image_name = image_name.replace("/" + DHIS2_DATA_IMAGE, "")
+    name_with_prefix = "{}-{}".format(PROJECT_NAME_PREFIX, clean_image_name)
+    return re.sub(r"[^\w]", "", name_with_prefix)
```

Comment:
Proposal: ``return re.sub(r"[^\w]", "-", name_with_prefix)``

Right now: eyeseetea/dhis2-data:2.30-sierra -> d2dockereyeseetea230sierra
Expected: eyeseetea/dhis2-data:2.30-sierra -> d2-docker-eyeseetea-2-30-sierra

Note: Here with prefix d2-docker

Reply:
Yep, that was what we had and was much nicer. Let's see if we can recover it.